### PR TITLE
Fix handling of exit statements in nested do loop

### LINKED ISSUE #1419: exit statements are not handled properly in nested do loops
```fortran
program main
    implicit none
    integer :: i, j
    i = 0
    j = 0
    outer : do
        i = i + 1
        inner : do
            j = j + 1
            if (j == 5) then
                print *, j
                exit inner
            end if
        end do inner
        print *, "out of inner loop"
        exit outer
    end do outer
    print *, "out of outer loop"
end program
```

> GFortran 

```console
$ gfortran a.f90 && ./a.out
           5
 out of inner loop
 out of outer loop
```

> LFortran

```console
$ lfortran a.f90
5
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
out of inner loop
```

LFortran generates incorrect LLVM which results in infinite looop to print `out of inner loop`.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #1419.

### CODE REVIEW

**Review by czgdp1807 — APPROVED** [review]

File: integration_tests/do_loop_02.f90 [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+            if (j == 5) then
+                print *, j
+                exit inner
+            end if
+        end do inner
+        print *, "out of inner loop"
+        exit outer
+    end do outer
+    print *, "out of outer loop"
+end program
```
Comment (certik) [inline integration_tests/do_loop_02.f90:19]: In another PR, let's test the exact result that "i" or "j" (or both) have to be, to ensure everything actually runs correctly.

**Review by certik — APPROVED** [review]

**Review by czgdp1807 — CHANGES_REQUESTED** [review]
I agree with https://github.com/lfortran/lfortran/pull/1420#discussion_r1132766982 and I request to do it here before this goes in. If you want to merge this urgently then let me know, I will approve in that case.

File: integration_tests/do_loop_02.f90 (hunk 1 above)
Comment (Pranavchiku) [reply integration_tests/do_loop_02.f90:23]: Added checks to test exact result.

**Review by czgdp1807 — APPROVED** [review]

**Comment by certik** [comment]
Thanks @Pranavchiku for adding the test. @czgdp1807 it's typically not urgent. One big advantage of doing it right away is that it gets done and we can forget about it. So probably a good strategy is to do small stuff like this right away, and only more time consuming stuff later, so that we don't drop the ball on the PR just because it is not perfect.

**Comment by czgdp1807** [comment]
Seems like with a stronger test modifications in fix were doable. So it was worth it here.

**Pull request merged**